### PR TITLE
docs: add information about debugging with Docker

### PR DESCRIPTION
**Pull request by codebytere**
What it says on the tin.

cc @jkleinsc

### CODE REVIEW

File: electron/debugging/docker.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Sometimes, you may experience test discrepancies between your local machine and the ones used to run CI, and so want to replicate those failures in an environment that most closely resembles them.
+
+On Linux, that's going to require you to run tests in a Docker container.
+
+1. Ensure you have Docker installed on your Linux machine.
+    * `$ sudo apt install docker.io`
+2. Ensure the Docker daemon is running 
+    * `$ sudo systemctl start docker`
+3. Find Electron's Docker image in our CircleCI configuration - it should look something like:
+    * `electronjs/build:3bba0fdee0d5650e751c2561f1806f9138dfe56a`
```
Comment (ckerr) [inline electron/debugging/docker.md]: This should have more steps -- where would an end user look for this, and how should they find it?

Also, might be worth spelling out that it's not always going to literally be that hash

File: electron/debugging/docker.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+3. Mount the Docker container:
```
Comment (ckerr) [inline electron/debugging/docker.md]: s/3/4/

File: electron/debugging/docker.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+Where `source=/path/to/electron-gn` should be replaced with the path to the directory on your computer that contains `src`, and `target=/tmp/electron-gn` is the point at which that directory will be mounted in the Docker container.
+
+After this command has finished executing, you should find yourself in the Docker container on the command line:
+
+```sh
+builduser@193defac9168:$
+```
+
+We chose to mount the Electron directory to `/tmp/electron-gn`, so we should now run:
```
Comment (ckerr) [inline electron/debugging/docker.md]: Should add for clarity when steps are to be run inside of Docker, or when they should be run in a native shell

File: electron/debugging/docker.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+```sh
+$ export DISPLAY=':99.0'
+$ sh -e /etc/init.d/xvfb start
+```
+
+From there, we should be good to go, and we can now run the tests inside the Docker shell:
+
+```sh
+$ node script/spec-runner --no-sandbox
```
Comment (jkleinsc) [inline electron/debugging/docker.md]: Why `--no-sandbox`?  This is only needed when running as root and you should be entering into the docker container with the `builduser` user.

Comment (jkleinsc) [inline electron/debugging/docker.md]: Why `--no-sandbox` here?  That is only needed if running as root, but the docker container should default to `builduser` which is a non root user

Comment (codebytere) [reply electron/debugging/docker.md]: ah you're right - i copied this over from my earlier test runs where i was root in my linux vm. I'll remove it.

**Pull request merged**